### ISSUE TITLE
Lucerne hammer can't be used to train hooked weapons or polearms skill, despite being both.

### ISSUE THREAD

**Issue by MrHrulgin** [CleverRaven/Cataclysm-DDA#69969]
### Describe the bug

A lucerne hammer can't be used to train hooked weapons or polearms skill, despite being both.

### Attach save file

N/A

### Steps to reproduce

Have a lucerne hammer, be unable to train.

### Expected behavior

I'd expect that it would be usable for training

### Screenshots

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/6834262/b0393368-f89d-4134-9756-1c940ad15f8c)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/6834262/2e62c62e-7b97-4251-9f45-74d16ece33e2)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/6834262/17d0b931-893a-4390-b38b-1a1523d896eb)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: 61cbf90 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Blaze Industries [blazeindustries],
    Mind Over Matter [mindovermatter],
    Translate Complex Dialogue [translate_dialogue],
    Alternative Map Key [alt_map_key],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

I think this (and #69968) are symptoms of a larger issue regarding melee weapon training as its currently implemented. With the need to add each item individually to the lists, it makes adding new items more of a pain, as they'll need to be added to each weapon training list individually. It might be better to do what's done with the ranged weapon, where training is based on them having a level of a quality that identifies the kind of weapon. This would let the practice be managed by the appropriate tag, and then practice can pull from the weapon type tags instead of trying to maintain individual lists of weapons for each type of training.

**Comment by Termineitor244** [comment]
We already have weapon categories for use, the problem is that you can't ask for weapon categories when making a recipe requirements, that and the fact that even if you could, certain weapons that could be used as weapon of a certain category, are... Not very good for practice, example being shivs/spears/etc that are too fragile to be used for an hour without a good chance of them breaking, so you would need to manually exclude those items when making the recipes use weapon categories.

**Comment by MrHrulgin** [comment]
> ...the problem is that you can't ask for weapon categories when making a recipe requirements... 

How do the gun training ones work then, with a tool quality applied to each one? Training pistol skill just requires that you have a gun with a pistol quality of 1. Being able to set the qualities of a weapon seems perhaps more scalable than editing the practice categories with each new weapon.

**Comment by Termineitor244** [comment]
> How do the gun training ones work then, with a tool quality applied to each one? Training pistol skill just requires that you have a gun with a pistol quality of 1. Being able to set the qualities of a weapon seems perhaps more scalable than editing the practice categories with each new weapon.

Tool qualities are not the same as weapon categories, it would be redundant to add for this tool qualities that are in essence the same as weapon categories (Something like what, knife quality?), the weapon categories in the first place were never planned for all the extra things it ended up being used for, the original concept was implemented to facilitate and equalize the burden in martial arts weapon lists, for mod inter-compatibility and intra-martial arts logic, nothing more and nothing less, most of them I added with that in mind, and the ones I did not where added with similar ideas, the mechanic was expanded later on for use in specific techniques, now more recently for specific proficiencies training (Live combat), but practice recipes is where they are still unable to be used.

About the rifles and pistols... No idea where their tool qualities originate from, Since I don't see any reference to those qualities in the code of the items, I think they are inferred automatically from their skills used... So this is not possible to be used for melee weapon categories which vary a lot between the 3 skills that govern their usage, and which use weapons that are widely different to use if even they are all under the same skill, the different kinds of swords are a good example of this.

I'm not suggesting that the current system is any good for combat practice recipes, but they are an anomaly in the first place, the combat category was not even contemplated when the practice recipes were first implemented, and weapon categories were introduced before them too, *combat practice* in general was at one time even strongly debated over how it could be implemented in the game, if at all, with one idea even being practicing with a real, dead body! I would know, since I was there in #51598. The fact that you can now even *train your specific proficiency* with a certain kind of weapon is already amazing, but the combat training and weapon categories we have were not designed with those in mind...

TLDR: The current system is already pretty good for what we had at first, so the fact that these can be made to work with minimal effort is pretty amazing already, and the options available to ranged weapons are not possible or desirable for use in these recipes.

I think the correct approach to this would be to allow the usage of weapon categories in recipe requirements/tools, though making the actual changes to the recipes would need some good eyes to look at them to avoid any problem with inadequate weapons.
